client/src/Components/NewQuestion.js

```diff
@@ -21,48 +21,211 @@ export default function NewQuestion() {
 			answer_a_correct: "false",
 			answer_b_correct: "false",
 			answer_c_correct: "false",
-			answer_d_correct: "true",
+			answer_d_correct: "false",
 			answer_e_correct: "false",
 			answer_f_correct: "false",
 		},
-		explanation: "rmdir deletes an empty directory",
+		explanation: "",
 		tip: null,
 		tags: [],
-		code: "linux",
+		code: "",
 		difficulty: "Easy",
 	});
 	const [md, setMd] = useState("");
+	const [cd, setCd] = useState("");
 	const textChanged = (e) => {
 		setMd(e.target.value);
+		setQuestionToSet({ ...questionToSet,
+			question: `${e.target.value}`,
+
+		},
+		);
+	};
+
+	const submitQuestion =()=>{
+		if(questionToSet.answers.answer_a.length<2){
+
+			alert("answer a should have at least 2sybols");
+
+			// } else if(newQuizzQuestions.length<5){
+
+			// 	alert("Quizz  should have at least 5 questions");
+
+		} else{
+			sendQestion();
+			alert("question submitted");
+		}
+	};
+	const sendQestion =()=>{
+
+		fetch("http://localhost:3100/api/question", { method:"POST",headers: { "Content-type": "application/json" },
+			body: JSON.stringify(questionToSet) })
+			.then((response) => response.json())
+			.then((data) =>console.log(data))
+			.catch((err) => console.error(err));
+
+
+
+	};
+	const codeChanged = (e) => {
+		setCd(e.target.value);
+		setQuestionToSet({ ...questionToSet,
+			code: `${e.target.value}`,
+
+		},
+		);
 	};
 	console.log( JSON.parse(JSON.stringify(Questions)));
 	const  answer_aChanged = (e) => {
 		setQuestionToSet({ ...questionToSet,
-			answers:{
+			answers:{ ...questionToSet.answers,
 				answer_a:`${e.target.value}`,
 			},
 		},
 		);
 	};
+	const  answer_bChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_b:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_cChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_c:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_dChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_d:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_eChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
```

Comment:
to prevent of repeat you can create a function for all of them

Reply:
i tried first but it gets complicated propose to leave for further development on later steps